audisp-tacplus/audisp-tacplus.c

```diff
@@ -93,190 +115,53 @@ hup_handler(int sig __attribute__ ((unused)))
-                        tac_srv_no < TAC_PLUS_MAXSERVERS;
-                        server = server->ai_next) {
-                        tac_srv[tac_srv_no].addr = server;
-                        /* use current key, if our index not yet set */
-                        if(tac_key_no && !tac_srv[tac_srv_no].key)
-                            tac_srv[tac_srv_no].key =
-                                strdup(tac_srv[tac_key_no-1].key);
-                        tac_srv_no++;
-                    }
-                }
-                else {
-                    syslog(LOG_ERR,
-                        "skip invalid server: %s (getaddrinfo: %s)",
-                        server_buf, gai_strerror(rv));
-                }
-            }
-            else {
-                syslog(LOG_ERR, "maximum number of servers (%d) exceeded, "
-                    "skipping", TAC_PLUS_MAXSERVERS);
-            }
-        }
-        else if(debug) /* ignore unrecognized lines, unless debug on */
-            syslog(LOG_WARNING, "%s: unrecognized parameter: %s",
-                progname, lbuf);
-    }
-
-    if(level == 0 && (!tac_service[0] || tac_srv_no == 0))
-        syslog(LOG_ERR, "%s version %d.%d.%d: missing tacacs fields in file %s, %d servers",
-            progname, _VMAJ, _VMIN, _VPATCH, configfile, tac_srv_no);
-
-    if(debug) {
-        int n;
-        syslog(LOG_NOTICE, "%s version %d.%d.%d tacacs service=%s", progname,
-            _VMAJ, _VMIN, _VPATCH, tac_service);
+    hup = 0;
 
-        for(n = 0; n < tac_srv_no; n++)
-            syslog(LOG_DEBUG, "%s: tacacs server[%d] { addr=%s, key='%s' }",
-                progname, n, tac_ntop(tac_srv[n].addr->ai_addr),
-                tac_srv[n].key);
-    }
+    connected_ok = 0; /*  reset connected state (for possible vrf) */
 
-    fclose(conf);
+    /* load config file: tacacs_config_file */
+    tacacs_ctrl = parse_config_file(tacacs_config_file);
 }
 
-
-static void
-reload_config(void)
+/*
+ * Get user name by UID, and return NULL when not found user name by UID.
+ * The returned username should be free by caller.
+ */
+char *lookup_logname(uid_t auid)
 {
-    int i, nservers;
-
-    hup = 0;
-
-    /*  reset the config variables that we use, freeing memory where needed */
-    nservers = tac_srv_no;
-    tac_srv_no = 0;
-    tac_key_no = 0;
-    vrfname[0] = '\0';
-    tac_service[0] = '\0';
-    tac_protocol[0] = '\0';
-    tac_login[0] = '\0';
-    debug = 0;
-    acct_all = 0;
-    tac_timeout = 0;
-
-    for(i = 0; i < nservers; i++) {
-        if(tac_srv[i].key) {
-            free(tac_srv[i].key);
-            tac_srv[i].key = NULL;
-        }
-        tac_srv[i].addr = NULL;
+    struct passwd *pws;
+    pws = getpwuid(auid);
+    if (pws == NULL) {
+        /* Failed to get user information. */
+        return NULL;
     }
-
-    connected_ok = 0; /*  reset connected state (for possible vrf) */
-
-    audisp_tacplus_config(configfile, 0);
```

Comment:
>audisp_tacplus_config [](http://example.com/codeflow?start=4&length=21)

Seem that you break upstream features, so there is no chance to upstream. Could you add new features without breaking old ones?